### PR TITLE
Game Commands: Place vehicle

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Seems to work. Could do with breaking up into sub functions.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Replays match.

File: src/OpenLoco/src/GameCommands/Vehicles/VehiclePlace.cpp [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    {
+        Vehicles::TrackAndDirection::_RoadAndDirection rtad(0, 0);
+        rtad._data = trackAndDirection;
+
+        if (rtad.isReversed())
+        {
+            rtad.setReversed(false);
+            auto& roadSize = World::TrackData::getUnkRoad(rtad._data);
+            pos -= roadSize.pos;
+            if (roadSize.rotationEnd < 12)
```
Comment (ZehMatt) [inline src/OpenLoco/src/GameCommands/Vehicles/VehiclePlace.cpp:29]: Is 12 just std::size(World::kRotationOffset), if yes we should probably not use a magic.

Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/Vehicles/VehiclePlace.cpp:29]: No it's something like not diagonal

**Review by AaronVanGeffen — COMMENTED** [review]
Nice work! This all appears to be working correctly. Agreed that the functions could do with some splitting up though. Could you see to that?

An oddity I noticed while testing is that ghost vehicles stay placed down when closing the vehicle window for the vehicle being placed. This also occurs on current master, but seems like the wrong behaviour. I'll file an issue.

**Comment by duncanspumpkin** [comment]
> Nice work! This all appears to be working correctly. Agreed that the functions could do with some splitting up though. Could you see to that?
> 
> An oddity I noticed while testing is that ghost vehicles stay placed down when closing the vehicle window for the vehicle being placed. This also occurs on current master, but seems like the wrong behaviour. I've filed this in #2526.

oh i thought fbbe64b76b14aa790cf927920159e589220bdc6a covered that. Its still quite long but its a more manageable chunk

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**